types/koa/index.d.ts

```diff
@@ -533,10 +535,12 @@ declare namespace Application {
         [key: string]: any;
     }
 
-    type Middleware<StateT = DefaultState, CustomT = DefaultContext> = compose.Middleware<
-        ParameterizedContext<StateT, CustomT>
+    type Middleware<StateT = DefaultState, ContextT = DefaultContext, ResponseBodyT = any> = compose.Middleware<
+        ParameterizedContext<StateT, ContextT, ResponseBodyT>
     >;
 
+    type Controller<ResponseBodyT = any, StateT = DefaultState, ContextT = DefaultContext> = (ctx: ParameterizedContext<StateT, ContextT, ResponseBodyT>) => void;
```

Comment:
I don't think it makes sense for this to become part of the Koa library types, within Koa, there isn't any reference to a `Controller`.

Reply:
I was wondering about the same. Indeed Koa is not using this terminology. Yet found it pretty useful, as most of the people will need that, but maybe you right it belongs more to the @koa/router...

I am totally fine reverting this change :)

Reply:
Removed `Controller`